patches/tModLoader/Terraria/ModLoader/ModPlants.cs

```diff
@@ -276,7 +276,7 @@ public abstract class ModPalmTree : ITree
 	/// <summary>
 	/// Used mostly for vanilla tree shake loot tables
 	/// </summary>
-	public virtual TreeTypes CountsAsTreeType => TreeTypes.Palm;
+	public virtual TreeTypes CountsAsTreeType => TreeTypes.ModPalmTree;
```

Comment:
This will need porting notes. Minor breakage for people relying on the default behavior. Also need to update the doc string for this member to mention the new hook.